resource_test.go

```diff
@@ -364,6 +368,58 @@ func TestGetResource(t *testing.T) {
 				End()
 		},
 	)
+
+	t.Run("can get all top rated resources", func(t *testing.T) {
+
+		client := testServer.Client()
+		uri := testServer.URL + resourceURI
+		request, err := http.NewRequest("GET", uri, nil)
+		if err != nil {
+			t.Log(err)
+		}
+		request.Header.Set("authorization", userToken)
+
+		response, err := client.Do(request)
+
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		var responseMap map[string]interface{}
+
+		dec := json.NewDecoder(response.Body)
+
+		dec.UseNumber()
+
+		dec.Decode(&responseMap)
+
+		expectedResourceTitle1 := "A top rated resource"
+		actualResourceTitle1 := responseMap["resources"].([]interface{})[0].(map[string]interface{})["Title"]
+
+		actualResourceTitle2 := responseMap["resources"].([]interface{})[0].(map[string]interface{})["Title"]
+
+		expectedRecommendations1 := int64(10)
+		actualRecommendations1 := responseMap["resources"].([]interface{})[0].(map[string]interface{})["Recommendations"]
+
+		actualRecommendations2 := responseMap["resources"].([]interface{})[1].(map[string]interface{})["Recommendations"]
+
+		aR1, _ := actualRecommendations1.(json.Number).Int64()
+
+		aR2, _ := actualRecommendations2.(json.Number).Int64()
+
+		if expectedRecommendations1 != aR1 {
+			t.Fatalf("Expected resource recommendations %v to equal %v", aR1, expectedRecommendations1)
+		}
+
+		if aR1 < aR2 {
+			t.Fatalf("Expected resource %s to come before resource %s", actualResourceTitle1, actualResourceTitle2)
+		}
+
+		if expectedResourceTitle1 != actualResourceTitle1 {
+			t.Fatalf("Expected resource title %s to equal %s", actualResourceTitle1, expectedResourceTitle1)
+		}
+
+	})
```

Comment:
About the tests, you would remember I updated the supertest we were using to be able to use the request to make assertions like this in all scenarios
```
Request(testServer.URL, t).
				Get(fmt.Sprintf("%v%v", resourceURI, publicResource.Id)).
				Set("authorization", userToken).
				Expect(200).
				Expect("Content-Type", "application/json").
				Expect(expectedResponse).
				End()
```
We can still merge this and later we'll have a task to harmonize every tests to follow that pattern. The advantage I was trying to get was to make the whole test structure simple and consistent; simulate expected response and just do a whole big assert. I'm not sure but it seems I made it in such a way that it will work for partial responses too.
I will take a look.